s3sync.go

```diff
@@ -253,15 +253,17 @@ func (m *Manager) syncS3ToLocal(ctx context.Context, chJob chan func(), sourcePa
 }
 
 func (m *Manager) copyS3ToS3(ctx context.Context, file *fileInfo, sourcePath *s3Path, destPath *s3Path) error {
-	println("Copying", file.name, "to", destPath.String())
+	copySource := sourcePath.bucket + "/" + sourcePath.bucketPrefix + "/" + file.name
+	destinationKey := destPath.bucketPrefix + "/" + file.name
```

Comment:
Seems doesn't work correctly when bucketPrefix is empty.
Fixing it and adding a test in https://github.com/seqsense/s3sync/pull/396

Reply:
https://github.com/seqsense/s3sync/actions/runs/9443166273/job/26006439773#step:5:231
```
2024/06/10 05:53:16 Copying from s3-source//bar/baz/README.md to key /bar/baz/README.md in bucket s3-destination
2024/06/10 05:53:16 Copying from s3-source//README.md to key /README.md in bucket s3-destination
2024/06/10 05:53:16 Copying from s3-source//foo/README.md to key /foo/README.md in bucket s3-destination
2024/06/10 05:53:16 Copying from s3-source/bar/baz/README.md to key hoge/baz/README.md in bucket s3-destination

--- FAIL: TestS3sync (0.08s)
    --- FAIL: TestS3sync/S3ToS3Copy (0.01s)
        s3sync_test.go:137: Sync should be successful NoSuchKey: The specified key does not exist.
            	status code: 404, request id: e502a7fa-7c82-4c7a-8c73-83a8f2c3e9cb, host id: s9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=
            NoSuchKey: The specified key does not exist.
            	status code: 404, request id: 109b4e6e-4db2-4d9b-b5de-d1bdffd3373f, host id: s9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=
            NoSuchKey: The specified key does not exist.
            	status code: 404, request id: a67aa2e9-b696-4996-bfa1-089c0dc4f8b1, host id: s9lzHYrFp76ZVxRcpX9+5cjAnEH2ROuNkd2BHfIa6UkFVdtjf5mKR3/eTPFvsiP/XV/VLi31234=
```